app/src/main/java/org/greenstand/android/TreeTracker/permission/Permissions.kt

```diff
@@ -0,0 +1,174 @@
+                        // TODO Check if location is enabled from the LocationUpdateManager before calling enableLocation
+                        return
+                    }
+                    (perm.shouldShowRationale || !perm.permissionRequested) -> {
+                        AlertDialog(
+                            onDismissRequest = { navController.popBackStack() },
+                            title = {
+                                Text(text = stringResource(R.string.accept_location_permission_header)
+                                )
+                            },
+                            text = {
+                                Text(
+                                    text = stringResource(R.string.accept_location_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            })
+                    }
+
+                }
+            }
+            Manifest.permission.ACCESS_COARSE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        // TODO Check if location is enabled from the LocationUpdateManager before calling enableLocation
+                        return
+                    }
+                    (perm.shouldShowRationale || !perm.permissionRequested) -> {
+                        AlertDialog(
+                            onDismissRequest = { navController.popBackStack() },
+                            title = {
+                                Text(text = stringResource(R.string.accept_location_permission_header))
+                            },
+                            text = {
+                                Text(
+                                    text = stringResource(R.string.accept_location_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()

```

Comment:
Why does this exit the screen? Seems to make more sense to request permission on camera screen like it initially did.

Reply:
It doesn't actually exit the screen but reloads the camera screen as the camera was initially called without permission and would show blank.